### PR TITLE
Fix Crash in AbstractChangeImplementationCodeRefactoringProvider

### LINKED ISSUE #70232: `AbstractChangeImplementationCodeRefactoringProvider` crashes with `InvalidOperationException` calling LINQ's `First`
```
System.InvalidOperationException : Sequence contains no elements
   at System.Linq.Enumerable.First[TSource](IEnumerable`1 source)
   at System.Linq.ImmutableArrayExtensions.First[T](ImmutableArray`1 immutableArray)
   at async Microsoft.CodeAnalysis.CSharp.ImplementInterface.AbstractChangeImplementationCodeRefactoringProvider.ComputeRefactoringsAsync(<Unknown Parameters>)
   at System.Runtime.ExceptionServices.ExceptionDispatchInfo.Throw()
   at async Microsoft.CodeAnalysis.CodeRefactorings.CodeRefactoringService.GetRefactoringFromProviderAsync(<Unknown Parameters>)
```

Minimal repro:

```csharp
interface I
{
    event EventHandler Click;
}

class C : I
{
    event EventHandler I.Click { add { } remove { } }

    event EventHandler I.Click
}
```

### PR DESCRIPTION
**Pull request by akhera99**
Fixes #70232
Assumption was made that implementations would exist -- not true in the case of an error.

### CODE REVIEW

**Review by dibarbet — COMMENTED** [review]
Can we add a test?  Probably here https://github.com/dotnet/roslyn/blob/main/src/Features/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs

File: src/Features/CSharp/Portable/ImplementInterface/AbstractChangeImplementationCodeRefactoringProvider.cs [hunk 1]
```diff
@@ -71,8 +71,11 @@ public sealed override async Task ComputeRefactoringsAsync(CodeRefactoringContex
             // (like "GetEnumerator"), instead of the auto-generated names that the compiler makes
             // like: "System.IEnumerable.GetEnumerator"
             directlyImplementedMembers.AddRange(member, member.ExplicitOrImplicitInterfaceImplementations());
+            var implementations = member.ExplicitOrImplicitInterfaceImplementations();
+            if (implementations.IsEmpty)
+                return;
```
Comment (Youssef1313) [inline src/Features/CSharp/Portable/ImplementInterface/AbstractChangeImplementationCodeRefactoringProvider.cs:76]:
```suggestion
            var implementations = member.ExplicitOrImplicitInterfaceImplementations();
            if (implementations.IsEmpty)
                return;

            directlyImplementedMembers.AddRange(member, implementations);
```

Comment (Youssef1313) [inline src/Features/CSharp/Portable/ImplementInterface/AbstractChangeImplementationCodeRefactoringProvider.cs:76]: Ideally, I think the refactoring should still be offered for this error scenario. But that scenario isn't very important so if that would be hard to support I think it's not a problem.

The confusing part here is that `ExplicitOrImplicitInterfaceImplementations` is empty, while `ExplicitInterfaceImplementations` is non empty:

https://github.com/dotnet/roslyn/blob/7eff5ed876f1d5f64dab9eb1b7d1c22e6ee587fa/src/Features/CSharp/Portable/ImplementInterface/CSharpImplementImplicitlyCodeRefactoringProvider.cs#L39-L41

Comment (genlu) [reply src/Features/CSharp/Portable/ImplementInterface/AbstractChangeImplementationCodeRefactoringProvider.cs:76]: nit: declaration of 'directlyImplementedMembers' could be moved after the return too

File: src/Features/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs [hunk 2]
```diff
@@ -11871,5 +11871,37 @@ IEnumerator IEnumerable.GetEnumerator()
                 """,
             }.RunAsync();
         }
+
+        [Fact, WorkItem("https://github.com/dotnet/roslyn/issues/70232")]
+        public async Task TestMissingWhenAlreadyContainingImpl()
```
Comment (Youssef1313) [inline src/Features/CSharpTest/ImplementInterface/ImplementInterfaceTests.cs:11876]: Shouldn't this be in https://github.com/dotnet/roslyn/blob/main/src/Features/CSharpTest/ImplementInterface/ImplementImplicitlyTests.cs ?

**Review by dibarbet — APPROVED** [review]

**Pull request merged**